Slim/Player/Protocols/HTTP.pm

```diff
@@ -1105,11 +1107,11 @@ sub getMetadataFor {
 		}
 	}
 	else {
+		# make sure that protocol handler is what the $song wanted, not just the $url-based one
+		my $handler = $forPlaying ? $song->currentTrackHandler : Slim::Player::ProtocolHandlers->handlerForURL($url);
```

Comment:
This is an important one if we want to make sure that the custom PH for the url track being played (the song) is called, even if the url is using a different PH